nomenclature/testing.py

```diff
@@ -36,11 +36,14 @@ def assert_valid_structure(path: Path):
         :class:`DataStructureDefinition`
       - If a `mappings` folder exists, it must be a valid :class:`RegionProcessor`
     """
-    definition = nomenclature.DataStructureDefinition(path / "definitions")
-    if (path / "mappings").is_dir():
+
+    definition = nomenclature.DataStructureDefinition(path / str(definitions))
+    if (path / str(mappings)).is_dir():
         nomenclature.RegionProcessor.from_directory(
-            path / "mappings"
+            path / str(mappings)
```

Comment:
Since we expect a string input anyway I don't think this is needed
```suggestion
            path / mappings
```
should be enough